File: removeDuplicates.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int unique = 0;
        for (int i = unique; i < nums.length; i++) {
            if (nums[unique] != nums[i]) {
                unique++;
                nums[unique] = nums[i];
            }
        }
        return unique + 1;
    }
}

File: removeDuplicates.py
# Remove duplicate values in an array, without using any remove/pop/etc functions
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        num = set(nums)
        k = len(num)
        j = 1

        while j < k:
            for i in range(len(nums)-1):
                # if the next element isn't equal to the current element (unique so far)
                if nums[i] != nums[i+1]:
                    # set that new value to the jth index in the array
                    nums[j] = nums[i+1]
                    j += 1
        return k
